### PR TITLE
Simplify UnitTest file setup

### PR DESCRIPTION
**Pull request by scannillo**
### Reason for changes

In one of our previous PRs, the testing structure / directory setup got wonky. We had duplicate directories for each test module in Finder that didn't correspond correctly in Xcode. 

Also, we had files floating around that are actually shared b/w multiple test targets (ex: networking mocks for CardClient and PaymentsCore).

### Changes

- Resolves the duplicates and nesting issues. 
- Adds a `TestShared` framework which we can use to house shared mocks b/w test modules.
  - Example, `CardTest` and `CoreTests` both need `URLSessionMocks`.
  - This also helps visualize files & tests specific to that module.
- Reduce the number of nested directories.
- Gives nested directories more descriptive names.

### Before
Screen Shot 2021-10-06 at 3 13 23 PM

### After

Screen Shot 2021-10-06 at 3 50 09 PM

### Note

When adding test files, or new files in general, it is always worth double checking in the Build Settings that the file you created belongs to the proper target, or test target. Even though Xcode might show it nested in a certain place, sometimes it links the file in the wrong way. _Xcode gonna Xcode._

### Authors
@scannillo

### CODE REVIEW

File: Tests/NetworkingMocks/MockResponse.swift [hunk 1]
```diff
@@ -1,9 +0,0 @@
-import Foundation
-
-protocol MockResponse {
```
Comment (scannillo) [inline Tests/NetworkingMocks/MockResponse.swift:3]: This was a duplicate file.

File: Tests/NetworkingMocks/Mocks/MockRequestResponse.swift [hunk 2]
```diff
@@ -1,66 +0,0 @@
-import Foundation
-import PaymentsCore
-
-class MockRequestResponse<Response: Codable>: APIRequest, MockResponse {
```
Comment (scannillo) [inline Tests/NetworkingMocks/Mocks/MockRequestResponse.swift:4]: This was a duplicate file.

File: Tests/UnitTests/CardTests/CardTests.swift [hunk 3]
```diff
@@ -1,12 +0,0 @@
-import XCTest
-@testable import Card
-
-final class CardTests: XCTestCase {
```
Comment (scannillo) [inline Tests/UnitTests/CardTests/CardTests.swift:4]: Replaced with `Card_Tests`, and updated appropriately. This wasn't compiling on master - I think it was a floating file / not actually running in any test suite.

File: UnitTests/CardTests/Card_Tests.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import XCTest
+@testable import Card
+
+final class Card_Tests: XCTestCase {
+    func testCard_setsProperExpiryStringFormat() {
```
Comment (scannillo) [inline UnitTests/CardTests/Card_Tests.swift:5]: Fixes issue where this file wasn't actually compiling / running in the previous test suite.

**Review by sshropshire — APPROVED** [review]
🌊

**Review by minhthenguyen — APPROVED** [review]
This looks great!!! thank you for doing this!!! :rocket:

**Review by jaxdesmarais — APPROVED** [review]
🚢  this looks great!

**Pull request merged**